explainer.md

```diff
@@ -68,356 +107,405 @@ If the set of pages being badged corresponds to an [Installed Web Application](h
-  length, as shown/described
-  [here](https://eternalstorms.wordpress.com/2016/10/29/how-to-badge-an-apps-icon-in-the-dock/).
-  The API is
-  [`NSDockTile.setBadgeLabel`](https://developer.apple.com/documentation/appkit/nsdocktile/1524433-badgelabel).
-* Flag badges are passed as an empty string to `setBadgeLabel` (if that works,
-  otherwise pass a space character). This shows an empty red circle.
-* Integer badges are saturated to 3 or 4 digits (with a "+" if overflowing), and
-  passed as a string to `setBadgeLabel`.
+* The URL scoping complexity only applies to the "handle" API.
+* The push messaging complexity only applies to the "handle" API.
+* We wouldn't need `canBadgeDocument`; the presence of the "document" API itself
+  would indicate whether the tab will be badged in the UI.
 
-### Universal Windows Platform
+Other reasons to separate:
 
-Requires Windows 10 or higher, and requires that the user agent be a "UWP" app.
-Note that Google Chrome and Mozilla Firefox do not fall into this category, and
-hence don't have access to the necessary APIs.
+* Sites that only want to badge their app icon or only want to badge their tab
+  could just concern themselves with the relevant API.
+* The "document" API could potentially support badges that aren't supported by
+  host operating systems, such as arbitrary Unicode characters.
 
-* The host API is
-  [`BadgeNotification`](https://docs.microsoft.com/en-us/uwp/api/windows.ui.notifications.badgenotification),
-  which allows either a positive integer (saturated at 99) or a fixed glyph
-  (shown
-  [here](https://docs.microsoft.com/en-us/windows/uwp/design/shell/tiles-and-notifications/badges)).
-* Integer badges are just passed straight through into this API.
-* Flag badges are passed as a fixed glyph, perhaps "attention", since I don't
-  think there is a way to show an empty circle.
+## Detailed API proposal
 
-This will show up on both the Taskbar and Start Menu tile.
-
-### Legacy Windows (Win32)
-
-Requires Windows 7 or higher.
-
-* The host API is
-  [`ITaskbarList3::SetOverlayIcon`](https://docs.microsoft.com/en-au/windows/desktop/api/shobjidl_core/nf-shobjidl_core-itaskbarlist3-setoverlayicon),
-  which allows applications to set a 16x16-pixel overlay icon in the corner of
-  the application's main icon in the Taskbar.
-* Due to the nature of being a 16x16-pixel icon, the user agent must render the
-  text or number into an image. It pretty much has to be 1–2 characters.
-* Flag badges are just passed as a coloured circle.
-* Integer badges are rendered as the number, if a single digit, or "+", if
-  greater than 9.
+### The model
 
-Note: This API only allows badging a currently open window.
+A badge is associated with a [scope](https://www.w3.org/TR/appmanifest/#navigation-scope).
 
-### Android
+* Documents are badged with the most specific badge for their URL (i.e. prefer a badge for `/page/1` to a badge for `/page/` when on the url `/page/1?foo=7`).
```

Comment:
This is already discussed in the [Nested Scopes](#nested-scopes) section.

Reply:
That's OK. It needs to be summarized in the model. In some sense, this entire section is just a summary of what's been discussed above, showing the whole API surface in one place.